frontend/src/components/BarChartPreview.tsx

```diff
@@ -86,18 +90,13 @@ const BarChartPreview = (props: Props) => {
             <YAxis
               dataKey={props.bars.length ? props.bars[0] : ""}
               type={yAxisType()}
-              width={
-                (props.data
-                  ?.map(
-                    (d) => (d as any)[props.bars.length ? props.bars[0] : ""]
-                  )
-                  .map((c) => (c as string).length)
-                  .reduce((a, b) => (a > b ? a : b), 0) || 0) *
-                  8 +
-                24
-              }
+              width={Math.min(
+                UtilsService.getLargestHeader(props.bars, props.data) * 8 + 24,
+                220
+              )}
```

Comment:
It feels like lately we have been adding a lot of hardcoded "magical" numbers on the Chart components. Is there any way we can make this better?

Reply:
Yeah, I agree, those are kind of magic numbers, I had to spend a lot of time figuring that out, and I dont think it is perfect yet. The problem is recharts does not support much functionalities, so we need to do the calculations manually. One way I see we could do it better is to create variables for each number describing what they represent (offset, pixel by character, etc)